docs: advice for inline disabling of rules (#17458)
eslint/eslint: docs/src/use/configure/rules.md

## Disabling Rules

### Using configuration comments

> * Use with Caution
>     * Disabling ESLint rules inline should be restricted and used only in situations with a clear and
>       valid reason for doing so. Disabling rules inline should not be the default solution to resolve linting errors.

Reviewer: Can we reformat these bullets so that we have one bullet per point? Like this:

```suggestion
* **Use with caution. ** Disabling ESLint rules inline should be restricted and used only in situations with a clear and
      valid reason for doing so. Disabling rules inline should not be the default solution to resolve linting errors.
```

## Disabling Rules

> ### Use with Caution

Reviewer: I think we can use a bulleted list instead of subheadings and paragraphs for this content.

> Emphasize that disabling ESLint rules should be restricted and only in situations with a clear and valid reason for
> doing so. Disabling rules should not be the default solution to resolve linting errors.

Reviewer: ```suggestion
Disabling ESLint rules inline should be restricted and used only in situations with a clear and valid reason for doing so. Disabling rules inline should not be the default solution to resolve linting errors.
```

### Document the Reason

Developers should provide a comment explaining the reason for disabling a particular rule. This documentation should
clarify why the rule is being disabled and why it is necessary in that specific situation.

### Temporary Solutions

If a disabled comment is added as a temporary measure to address a pressing issue, developers should also create a
follow-up task or ticket to address the underlying problem adequately. This ensures that the disabled comment is
revisited and resolved at a later stage.

### Code Reviews and Pair Programming

Encourage team members to review each other's code regularly. Code reviews can help identify the reasons behind
disabling comments and ensure that they are used appropriately.

### Configurations

Whenever possible, prefer using ESLint configuration settings over disabled comments. Configuration files allow for
consistent and project-wide rule handling.

### Selective Rule Disabling

> If multiple rules need to be disabled, consider using block-level or file-level disable comments instead of disabling

Reviewer: As mentioned in the issue, this isn't the advice we want to give.

Author: ok removing this section

### Using configuration comments

> * **Use with Caution**: Disabling ESLint rules inline should be restricted and used only in situations with a clear and
>       valid reason for doing so. Disabling rules inline should not be the default solution to resolve linting errors.
> * **Document the Reason**: Developers should provide a comment explaining the reason for disabling a particular rule. This
>       documentation should clarify why the rule is being disabled and why it is necessary in that specific situation.
> * **Temporary Solutions**: If a disabled comment is added as a temporary measure to address a pressing issue, developers
>       should also create a follow-up task or ticket to address the underlying problem adequately. This ensures that the
>       disabled comment is revisited and resolved at a later stage.
> * **Code Reviews and Pair Programming**: Encourage team members to review each other's code regularly. Code reviews can help
>       identify the reasons behind disabling comments and ensure that they are used appropriately.
> * **Configurations**: Whenever possible, prefer using ESLint configuration settings over disabled comments. Configuration
>       files allow for consistent and project-wide rule handling.
> 

Reviewer: ```suggestion
* **Use with Caution.** Disabling ESLint rules inline should be restricted and used only in situations with a clear and
      valid reason for doing so. Disabling rules inline should not be the default solution to resolve linting errors.
* **Document the Reason.** Provide a comment explaining the reason for disabling a particular rule after the `--` section of the comment. This
      documentation should clarify why the rule is being disabled and why it is necessary in that specific situation.
* **Temporary Solutions.** If a disable comment is added as a temporary measure to address a pressing issue, create a follow-up task to address the underlying problem adequately. This ensures that the
      disable comment is revisited and resolved at a later stage.
* **Code Reviews and Pair Programming.** Encourage team members to review each other's code regularly. Code reviews can help
      identify the reasons behind disable comments and ensure that they are used appropriately.
* **Configurations.** Whenever possible, prefer using ESLint configuration files over disable comments. Configuration
      files allow for consistent and project-wide rule handling.

```

### Using configuration comments

* Use with Caution
    * Disabling ESLint rules inline should be restricted and used only in situations with a clear and
      valid reason for doing so. Disabling rules inline should not be the default solution to resolve linting errors.
* Document the Reason
    * Developers should provide a comment explaining the reason for disabling a particular rule. This
      documentation should clarify why the rule is being disabled and why it is necessary in that specific situation.
* Temporary Solutions

>     * If a disabled comment is added as a temporary measure to address a pressing issue, developers

Reviewer 2: By _disabled comment_ I think we mean `eslint-disable` comment?

#### Comment descriptions

> In cases where a specific rule consistently causes issues that are difficult to resolve, consider creating an exemption for that rule in the project's ESLint configuration with appropriate justifications.

Reviewer: This is a bit vague and doesn't use language that is familiar to ESLint users. What is an "exemption"? Are you saying to modify the rule options in the configuration file? If so, we should just say that. Also note that configuration files don't support reasons for disabling comments.
